app_feup/lib/view/Widgets/NavigationDrawer.dart

```diff
@@ -1,70 +1,84 @@
 import 'package:app_feup/view/Theme.dart';
 import 'package:flutter/material.dart';
 
+import '../../Main.dart';
 
 class NavigationDrawer extends StatefulWidget {
   final BuildContext parentContext;
 
-  NavigationDrawer({
-    @required this.parentContext
-  }){}
+  NavigationDrawer({@required this.parentContext}) {}
+
   @override
   State<StatefulWidget> createState() {
     return new NavigationDrawerState(parentContext: parentContext);
   }
 }
 
 class NavigationDrawerState extends State<NavigationDrawer> {
-
   final BuildContext parentContext;
 
-  NavigationDrawerState({
-    @required this.parentContext
-  }){}
+  NavigationDrawerState({@required this.parentContext}) {}
 
-  static final drawerItems = [
-    "Área Pessoal",
-    "Horário",
-    "Mapa de Exames",
-    "Sobre",
-    "Bug Report"
-  ];
+  Map drawerItems = {};
 
-  getCurrentRoute() => ModalRoute.of(parentContext).settings.name == null ? 
-  drawerItems[0]:
-  ModalRoute.of(parentContext).settings.name.substring(1);
+  @override
+  void initState() {
+    super.initState();
+
+    drawerItems = {
+      NAV_PERSONAL_AREA: _onSelectItem,
+      NAV_SCHEDULE: _onSelectItem,
+      NAV_EXAMS: _onSelectItem,
+      NAV_ABOUT: _onSelectItem,
+      NAV_BUG_REPORT: _onSelectItem,
+    };
+  }
 
-  _onSelectItem(int index) {
+  // Callback Functions
+  getCurrentRoute() => ModalRoute.of(parentContext).settings.name == null
+      ? drawerItems.keys.toList()[0]
```

Comment:
drawerItems[0] also does the job in that case I believe

Reply:
I think it is correct, mostly because drawerItems is a map and keys and values in maps can assume multiple types, using drawerItems[0] would try to access a value instead of the first key as expected, therefore always returning null for this scenario (I took your word into consideration and I tested it myself)

Reply:
Alright, will take your word for it!